src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -8,29 +8,63 @@
 using System.IO;
 using UnityEngine.SceneManagement;
 using System.Reflection;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Runtime.CompilerServices;
 #if UNITY_EDITOR
 using UnityEditor;
 #endif
 
 namespace Microsoft.MixedReality.SpectatorView
 {
+    [Serializable]
+    internal struct NameEntry
+    {
+        public string Name;
+        public AssetId[] Ids;
+
+        public NameEntry(string name, AssetId[] ids)
+        {
+            this.Name = name;
+            this.Ids = ids;
+        }
+
+        public override string ToString()
+        {
+            return $"{Name}";
+        }
+    }
+
     internal abstract class AssetCache : ScriptableObject
     {
         const string assetCacheDirectory = "Generated.StateSynchronization.AssetCaches";
+        protected const string assetsFolderName = "AssetCacheContent";
 
         public static TAssetCache LoadAssetCache<TAssetCache>()
             where TAssetCache : AssetCache
         {
             using (StateSynchronizationPerformanceMonitor.Instance.IncrementEventDuration(typeof(TAssetCache).Name, "LoadAssetCache"))
             using(StateSynchronizationPerformanceMonitor.Instance.MeasureEventMemoryUsage(typeof(TAssetCache).Name, "LoadingAssets"))
             {
-                return Resources.Load<TAssetCache>(typeof(TAssetCache).Name);
+                var cache = Resources.Load<TAssetCache>(typeof(TAssetCache).Name);
+
+                if (BroadcasterSettings.IsInitialized &&
+                    BroadcasterSettings.Instance.ForceLoadAllAssetsDuringInitialization)
+                {
+                    cache.LoadAllAssets();
+                }
+
+                return cache;
             }
         }
 
-        public static string GetAssetPath(string assetName, string assetExtension)
+        public static string GetAssetCachePath(string assetName, string assetExtension)
         {
-            return $"Assets/{assetCacheDirectory}/Resources/" + assetName + assetExtension;
+            return $"Assets/{assetCacheDirectory}/Resources/{assetName}{assetExtension}";
+        }
+
+        public static string GetAssetCachesContentPath(string assetName, string assetExtension)

```

Comment:
Nit: Can you add some comments about what these functions do (i.e. especially the two of them together being so close, taking the same set of parameters, it's not clear as a reader when I would call one and when I'd call the other).

Especially because this is the base class for a lot of other stuff, I think it's important to have these sorts of things documented for posterity's sake.

I think what I'm really looking for is documentation about what a "content" path is and a "regular path" is.